sdk/core/azure-core-management/src/main/java/com/azure/core/management/implementation/LocationPollStrategy.java

```diff
@@ -72,11 +72,19 @@ PollStrategy initializeStrategy(RestProxy restProxy,
             this.methodParser = methodParser;
             return new LocationPollStrategy(this);
         }
+
+        URL getLocationUrl() {
+            return locationUrl;
+        }
+
+        void setLocationUrl(URL locationUrl) {
+            this.locationUrl = locationUrl;
+        }
```

Comment:
Don't change API for things unrelated to the issue you're working on - these kind of changes should be reverted.

Reply:
looks not related, but it is related XD

`locationURL` is a field that gets assigned only in constuctor (in its own class), so we ask for it to be final.
but then, since field has default accessor, another class can extend its class and directly access this field and have it mutated.

So, the options here are:
1.- Add a set method in the same class to make it clear about that field is meant to be mutated. Then checkstyle won't ask for it to be final (that's basically what I did).
2.- Make checkstyle ignore fields with default and protected access. So no need to update API at all.

IMO, I don't like having fields directly exposed when it is not an static field. I think having always setter and getter methods for fields and encapsulating is better for readability and future maintainance.

Maybe let's have a quick call to talk about this one. :)

Reply:
I would suggest going towards 2). I would like to log it so people can choose to fix or suppress, but I don't know if we can without it failing the build?

Reply:
log it as a warning maybe?

Reply:
I believe we fail in all cases.

Reply:
I have taken option 2 and ignore default and protected fields for now.
We can create a new PR if we want to warn people about this and share the list of issues.